task.go

```diff
@@ -294,14 +303,18 @@ func (t *task) Exec(ctx context.Context, id string, spec *specs.Process, ioCreat
 	}, nil
 }
 
-func (t *task) Pids(ctx context.Context) ([]uint32, error) {
+func (t *task) Pids(ctx context.Context) ([]ProcessInfo, error) {
 	response, err := t.client.TaskService().ListPids(ctx, &tasks.ListPidsRequest{
 		ContainerID: t.id,
 	})
 	if err != nil {
 		return nil, errdefs.FromGRPC(err)
 	}
-	return response.Pids, nil
+	var processList []ProcessInfo
+	for _, p := range response.Processes {
+		processList = append(processList, ProcessInfo(*p))
```

Comment:
This works?

Reply:
It works if they have the same underlying types but will break if either struct changes. Happy to change it to update field by field.

Reply:
I think with go 1.9 and the aliases introduction, the conversion are a bit smarter

Reply:
I'd probably change it to do field by field since the client could get out of sync with the server.  It will make it more robust in the future for clients.

Reply:
Still familiarizing myself with go so not sure if this was the best move

Reply:
ok cool! :)

Reply:
No problem.  I honestly never tried this before.

Reply:
This works and it is okay. The two types are the same size and field order.

This _will_ break clients when we add a field.

Reply:
Final thoughts?

Reply:
change it so we don't break clients